packages/query/src/block-processor.ts

```diff
@@ -273,7 +278,7 @@ export class BlockProcessor implements BlockProcessorInterface {
 
       const isLastBlockOfEpoch = !!compactBlock.epochRoot;
       if (isLastBlockOfEpoch) {
-        void this.handleEpochTransition(compactBlock.height, latestKnownBlockHeight);
+        await this.handleEpochTransition(compactBlock.height, latestKnownBlockHeight);
```

Comment:
Needed to `await` due to an unlikely, but possible, race condition. When we save a new epoch to the database during an epoch transition, we look in the database for the _previous_ latest known epoch. We grab its epoch index, increment it by 1, then use that index for the new epoch we're saving to the database.

If that took a long time, for some reason, and in the meantime the block processor had already gotten to the next epoch, we could have two epochs with the same epoch index.

This change makes sure that epoch indexes increment in the correct order.